model/task_lifecycle.go

```diff
@@ -2447,7 +2450,7 @@ func ResetTaskOrDisplayTask(ctx context.Context, settings *evergreen.Settings, t
 				return errors.Wrap(err, "marking display task for reset")
 			}
 		}
-		return errors.Wrap(checkResetDisplayTask(ctx, settings, &taskToReset), "checking display task reset")
+		return errors.Wrap(checkResetDisplayTask(ctx, settings, user, origin, &taskToReset), "checking display task reset")
```

Comment:
For display tasks, will we error saying "the user can't restart these tasks" only when trying to reset? This seems like a bad user experience, since the user won't receive this error (presumably it should happen when they try to mark the display task for reset, although i know we won't know the number we're restarting if we're only restarting failed -- maybe then we use the worst case? We could also just not error for display tasks and move this to future work)

Reply:
This is a good question, I filed [DEVPROD-6991](https://jira.mongodb.org/browse/DEVPROD-6991) to tackle these UX improvements.

Reply:
Sounds good -- I think we'd want to address this ahead of rollout, or not include task groups or display tasks (realized task groups are also a culprit here)